test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -885,8 +888,25 @@ public static String randomGeohash(int minPrecision, int maxPrecision) {
         return geohashGenerator.ofStringLength(random(), minPrecision, maxPrecision);
     }
 
-    public static String randomTestTransport() {
-        return randomBoolean() ? NioTransportPlugin.NIO_TRANSPORT_NAME : MockTcpTransportPlugin.MOCK_TCP_TRANSPORT_NAME;
+    private static final SetOnce<Boolean> useNio = new SetOnce<>();
```

Comment:
You're correct.

When I do this:

```
private static final boolean useNio = randomBoolean();
    
    public static String getTestTransportType() {
        return useNio ? NioTransportPlugin.NIO_TRANSPORT_NAME : MockTcpTransportPlugin.MOCK_TCP_TRANSPORT_NAME;
    }

    public static Class<? extends Plugin> getTestTransportPlugin() {
        return useNio ? NioTransportPlugin.class : MockTcpTransportPlugin.class;
    }
```

The build fails. I get a bunch of these exceptions:

```
ERROR   0.00s J0 | DataCountsTests.initializationError <<< FAILURES!
   > Throwable #1: java.lang.NoClassDefFoundError: org.elasticsearch.test.ESTestCase
   >    at java.lang.Class.getDeclaredMethods0(Native Method)
   >    at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
   >    at java.lang.Class.getDeclaredMethods(Class.java:1975)
   >    at com.carrotsearch.randomizedtesting.ClassModel$3.members(ClassModel.java:215)
   >    at com.carrotsearch.randomizedtesting.ClassModel$3.members(ClassModel.java:212)
   >    at com.carrotsearch.randomizedtesting.ClassModel$ModelBuilder.build(ClassModel.java:85)
   >    at com.carrotsearch.randomizedtesting.ClassModel.methodsModel(ClassModel.java:212)
   >    at com.carrotsearch.randomizedtesting.ClassModel.<init>(ClassModel.java:207)
   >    at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
```

And I also see this:

```
   > Throwable #1: java.security.AccessControlException: access denied ("org.elasticsearch.ThreadPermission" "modifyArbitraryThreadGroup")
   >    at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
   >    at java.security.AccessController.checkPermission(AccessController.java:884)
   >    at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
   >    at org.elasticsearch.SecureSM.checkThreadGroupAccess(SecureSM.java:194)
   >    at org.elasticsearch.SecureSM.checkAccess(SecureSM.java:141)
   >    at java.lang.ThreadGroup.checkAccess(ThreadGroup.java:315)
   >    at java.lang.ThreadGroup.getParent(ThreadGroup.java:167)
   >    at com.carrotsearch.randomizedtesting.RandomizedContext.context(RandomizedContext.java:240)
   >    at com.carrotsearch.randomizedtesting.RandomizedContext.current(RandomizedContext.java:134)
   >    at org.apache.lucene.util.LuceneTestCase.random(LuceneTestCase.java:746)
   >    at org.elasticsearch.test.ESTestCase.randomBoolean(ESTestCase.java:489)
   >    at org.elasticsearch.test.ESTestCase.<clinit>(ESTestCase.java:891)
   >    at java.lang.Class.forName0(Native Method)
   >    at java.lang.Class.forName(Class.java:348)
Completed [1/643] on J3 in 1.67s, 0 tests, 1 error <<< FAILURES!
```